core/modules/qdisp/QueryRequest.cc

```diff
@@ -414,20 +443,49 @@ void QueryRequest::_processData(JobQuery::Ptr const& jq, int blen, bool last) {
     // Get a copy of the shared buffer pointer so askForResponseDataCmd can be deleted.
     ResponseHandler::BufPtr bufPtr = _askForResponseDataCmd->getBufPtr();
     _askForResponseDataCmd.reset(); // No longer need it, and don't want the destructor calling _errorFinish().
+
+    // The buffer has 2 parts.
+    // - The first (bytes = blen - ProtoHeaderWrap::getProtheaderSize())
+    //   is the result associated with the previously received header.
+    // - The second is the header for the next message.
+    int protoHeaderSize = proto::ProtoHeaderWrap::getProtoHeaderSize();
+    int respSize = blen - protoHeaderSize;
+    ResponseHandler::BufPtr nextHeaderBufPtr =
+        make_shared<vector<char>>(bufPtr->begin() + respSize, bufPtr->end());
+
+    // Read the result
     bool largeResult = false;
     int nextBufSize = 0;
-    bool flushOk = jq->getDescription()->respHandler()->flush(blen, bufPtr, last, largeResult, nextBufSize);
+    bool flushOk = jq->getDescription()->respHandler()->flush(respSize, bufPtr, last,
+                                                              largeResult, nextBufSize);
+    if (last) {
+        throw Bug("_processData result had 'last' true, which cannot be allowed.");
```

Comment:
Explain to me why this can no longer be true. What prevents SSI from setting last? An explanatory comment here would be good.

Reply:
The 'last' must show up as part of reading a header, not a result. If last is true while reading in the result, something really bad happened. I'll add a comment.

Reply:
It also looks like some shadowing is happening.